internal/heartbeat/heartbeat.go

```diff
@@ -85,6 +90,42 @@ func (s *HeartbeatData) RetrieveInfo() models.Heartbeat {
 	return heartbeatInfo
 }
 
+func (s *HeartbeatData) buildHardwareInfo() *models.HardwareInfo {
+	var hardwareInfo *models.HardwareInfo
+	var err error
+	if s.previousMutableHardwareInfo == nil {
+		// Only send all Hw info (mutable + immutable) for the 1st heartbeat, then send only mutable hw info
+		hardwareInfo, err = s.hardware.GetHardwareInformation()
+		if err != nil {
+			log.Errorf("cannot get full hardware information. DeviceID: %s; err: %v", s.workloadManager.GetDeviceID(), err)
+		}
+		s.previousMutableHardwareInfo = s.hardware.CreateHardwareMutableInformation()
+		if err != nil {
+			log.Errorf("cannot get mutable hardware information. DeviceID: %s; err: %v", s.workloadManager.GetDeviceID(), err)
+		}
+	} else {
+
+		hardwareInfo = s.getMutableHardwareInfoDelta()
+	}
+	return hardwareInfo
+}
+
+func (s *HeartbeatData) getMutableHardwareInfoDelta() *models.HardwareInfo {
+	hardwareMutableInfo := s.hardware.CreateHardwareMutableInformation()
+	var hardwareInfo *models.HardwareInfo
+
+	if s.configManager.GetDeviceConfiguration().Heartbeat.HardwareProfile.Scope == SCOPE_DELTA {
+		log.Debugf("Checking if mutable hardware information change between heartbeat (scope = delta). DeviceID: %s", s.workloadManager.GetDeviceID())
+		hardwareInfo = hw.GetMutableHardwareInfoDelta(*s.previousMutableHardwareInfo, *hardwareMutableInfo)
+		s.previousMutableHardwareInfo = hardwareMutableInfo
+	} else {
```

Comment:
same here, I would remove this else.

Reply:
you would return in the if?
I like to keep a single return in my function, except when an error is found and that the workflow should be interrupted but it's not the case here

Reply:
I would refactor like this: 

```
func (s *HeartbeatData) getMutableHardwareInfoDelta() *models.HardwareInfo {
	hardwareMutableInfo := s.hardware.CreateHardwareMutableInformation()
	hardwareInfo := hardwareMutableInfo

	if s.configManager.GetDeviceConfiguration().Heartbeat.HardwareProfile.Scope == SCOPE_DELTA {
		log.Debugf("Checking if mutable hardware information change between heartbeat (scope = delta). DeviceID: %s", s.workloadManager.GetDeviceID())
		hardwareInfo = hw.GetMutableHardwareInfoDelta(*s.previousMutableHardwareInfo, *hardwareMutableInfo)
	}
	s.previousMutableHardwareInfo = hardwareMutableInfo
	return hardwareInfo
}
```

Reply:
Ah! Yes I see, it could even be 
```
func (s *HeartbeatData) getMutableHardwareInfoDelta() *models.HardwareInfo {
	hardwareInfo := s.hardware.CreateHardwareMutableInformation()

	if s.configManager.GetDeviceConfiguration().Heartbeat.HardwareProfile.Scope == SCOPE_DELTA {
		log.Debugf("Checking if mutable hardware information change between heartbeat (scope = delta). DeviceID: %s", s.workloadManager.GetDeviceID())
		hardwareInfo = hw.GetMutableHardwareInfoDelta(*s.previousMutableHardwareInfo, *hardwareInfo)
	}
	s.previousMutableHardwareInfo = hardwareMutableInfo
	return hardwareInfo
}
```